File: easy/php/FuncCall.php
<?php

require_once __DIR__ . '/cmd_exec/InterfaceImpl.php';
require_once __DIR__ . '/common/VulnChild.php';
require_once __DIR__ . '/iface/VulnRunner.php';

class FuncCall
{
    public function runByChild(string $cmd): string
    {
        return (new VulnChild())->run($cmd);
    }

    public function runByInterface(string $cmd): string
    {
        $runner = new InterfaceImpl();
        return $runner->run($cmd);
    }
}

File: easy/python/func_call.py
from common.vuln_child import VulnChild
from cmd_exec.interface_impl import InterfaceImpl

class FuncCall:
    @staticmethod
    def run_by_child(cmd):
        child = VulnChild()
        child.exec(cmd)

    @staticmethod
    def run_by_interface(cmd):
        runner = InterfaceImpl()
        runner.run(cmd)
